napari/layers/labels/labels.py

```diff
@@ -525,7 +537,37 @@ def paint(self, coord, new_label, refresh=True):
             slice_coord = tuple(slice_coord)
 
         # update the labels image
-        self.data[slice_coord] = new_label
+
+        if self._overwrite:
+            self.data[slice_coord] = new_label
+        else:
+            self.paint_without_overwrite(self.data, slice_coord, new_label, 0)
 
         if refresh is True:
             self.refresh()
+
+    def paint_without_overwrite(
+        self, array, slice_coord, new_label, dimension
+    ):
+        if dimension == len(slice_coord) - 1:
+            for coord in self.read_dimension(slice_coord, dimension):
+                if array[coord] == 0:
```

Comment:
We should probably add a new property `self.background_label = 0` and use that here instead. Will be useful in case we expand to support colormaps with non-zero backgrounds at some point (some people use `-1` in the community). Will also make support features like an "eraser" more clear in the future as then it will pull the background label directly